src/ion/process.rs

```diff
@@ -99,17 +101,18 @@ impl<'a, F: Function> Env<'a, F> {
 
             let mut skips = 0;
             'alloc: loop {
-                trace!("  -> PReg range {:?}", preg_range_iter.peek());
+                let Some(next) = preg_range_iter.as_slice().first() else {
```

Comment:
Why was this moved to the top of the loop? It looks like there's some additional reworking of the algorithm here but I'd like to avoid semantic changes like this when in a PR that adds allocator support.